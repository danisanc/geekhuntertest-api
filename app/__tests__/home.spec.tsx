import renderer from 'react-test-renderer';

import Home from '../pages/index';

it('renders home page correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});