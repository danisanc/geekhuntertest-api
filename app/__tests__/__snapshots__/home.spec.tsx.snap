// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page correctly 1`] = `
<main
  className="max-w-3xl w-full mx-auto p-4 md:p-0"
>
  <h1
    className="text-2xl font-bold my-8"
  >
    Encontre o candidato ideal
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="grid md:grid-cols-2 grid-cols-1 gap-4 mb-8"
    >
      <div
        className="w-full"
      >
        <label
          className="block mb-2"
          htmlFor="city"
        >
          Cidade
        </label>
        <div
          className=" css-2b097c-container"
          id="city"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1wa3eu0-placeholder"
              >
                Selecione...
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-city-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="city"
            type="hidden"
            value=""
          />
        </div>
      </div>
      <div
        className="w-full"
      >
        <label
          className="block mb-2"
          htmlFor="techs"
        >
          Competencias
        </label>
        <div
          className=" css-2b097c-container"
          id="techs"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1wa3eu0-placeholder"
              >
                Selecione...
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-techs-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="techs"
              type="hidden"
            />
          </div>
        </div>
      </div>
      <div
        className="w-full"
      >
        <label
          className="block mb-2"
          htmlFor="minExp"
        >
          Experiencia minima
        </label>
        <input
          className="w-full p-1.5 rounded border border-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-500 transition"
          data-testid="input"
          id="minExp"
          name="minExp"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="w-full"
      >
        <label
          className="block mb-2"
          htmlFor="maxExp"
        >
          Experiencia maxima
        </label>
        <input
          className="w-full p-1.5 rounded border border-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-500 transition"
          data-testid="input"
          id="maxExp"
          name="maxExp"
          onChange={[Function]}
          type="number"
        />
      </div>
    </div>
    <button
      className="w-full p-2 bg-blue-500 text-white rounded"
      data-testid="submit"
      disabled={false}
      type="submit"
    >
      Buscar
    </button>
  </form>
  <section
    className="w-full py-8 space-y-4"
  />
</main>
`;
